src/inmanta/agent/handler.py

```diff
@@ -655,6 +655,51 @@ def do_changes(self, ctx: HandlerContext, resource: resources.Resource, changes:
         """
         raise NotImplementedError()
 
+    def deploy(
```

Comment:
I think is would be good to migrate to a module where deploy and dryrun are separate methods, also at this level, to make the dryrun guaranteed to be deploy-free.

Reply:
Fixed.